validator/proposer/service.go

```diff
@@ -44,15 +69,48 @@ func (p *Proposer) Stop() error {
 }
 
 // run the main event loop that listens for a proposer assignment.
-func (p *Proposer) run(done <-chan struct{}) {
+func (p *Proposer) run(done <-chan struct{}, client pb.ProposerServiceClient) {
+	sub := p.assigner.ProposerAssignmentFeed().Subscribe(p.assignmentChan)
+	defer sub.Unsubscribe()
 	for {
 		select {
 		case <-done:
 			log.Debug("Proposer context closed, exiting goroutine")
 			return
-		case <-p.beaconService.ProposerAssignment():
+
+		// TODO: On the beacon node side, calculate active and crystallized and update the
+		// active/crystallize state hash values in the proposed block.
+
+		// When we receive an assignment on a slot, we leverage the fields
+		// from the latest canonical beacon block to perform a proposal responsibility.
+		case latestBeaconBlock := <-p.assignmentChan:
 			log.Info("Performing proposer responsibility")
-			continue
+
+			// Extract the hash of the latest beacon block to use as parent hash in
+			// the proposal.
+			data, err := proto.Marshal(latestBeaconBlock)
+			if err != nil {
+				log.Errorf("Could not marshal latest beacon block: %v", err)
+				continue
+			}
+			latestBlockHash := blake2b.Sum512(data)
+
+			// TODO: Implement real proposals with randao reveals and attestation fields.
+			req := &pb.ProposeRequest{
+				ParentHash:              latestBlockHash[:],
+				SlotNumber:              latestBeaconBlock.GetSlotNumber() + 1,
+				RandaoReveal:            []byte{},
+				AttestationBitmask:      []byte{},
+				AttestationAggregateSig: []uint32{},
+				Timestamp:               ptypes.TimestampNow(),
+			}
+
+			res, err := client.ProposeBlock(p.ctx, req)
```

Comment:
Shouldn't we check if this is the correct slot window before we propose a block? It's probably out the scope for this PR, but we should  add a todo

Reply:
We do in the routine that listens for beacon blocks. We check if we are in the correct responsibility and slot assignment.